### PR TITLE
refactor: use fc-go-sdk vsock module

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
#578 

*Description of changes:*
Use the vsock module that was refactored into the firecracker-go-sdk.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kzys** [comment]
The main branch has some fixes regarding our test stability. Can you rebase your branch?

**Comment by austinvazquez** [comment]
> The main branch has some fixes regarding our test stability. Can you rebase your branch?

Rebasing off main instead of containerd-1.6 branch.

**Comment by kzys** [comment]
You need to remove https://github.com/firecracker-microvm/firecracker-containerd/blob/main/internal/vm/vsock_test.go.

File: internal/vm/vsock.go [hunk 1]
```diff
@@ -140,7 +32,7 @@ func VSockDialConnector(timeout time.Duration, udsPath string, port uint32) IOCo
 			timeoutCtx, cancel := context.WithTimeout(procCtx, timeout)
 			defer cancel()
 
-			conn, err := VSockDial(timeoutCtx, logger, udsPath, port)
+			conn, err := vsock.Dial(timeoutCtx, logger, udsPath, port)
```
Comment (kzys) [inline internal/vm/vsock.go:35]: Can you upgrade the SDK to the latest version and use DialContext here instead? It is needed to have https://github.com/firecracker-microvm/firecracker-go-sdk/pull/390.

**Review by kzys — CHANGES_REQUESTED** [review]

**Review by Kern-- — APPROVED** [review]

**Pull request merged**